File: reuse_java/src/test/java/com/kmmania/ThrowerTest.java
package com.kmmania;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.*;

public class ThrowerTest {

    @Test
    @DisplayName("display() method should display correct information")
    void displayShouldDisplayCorrectInformation() {
        // Given
        String firstName = "Alice";
        String lastName = "Smith";
        String speciality = "Throwing";
        Athlete mockAthlete = mock(Athlete.class);
        when(mockAthlete.mySpeciality()).thenReturn("My speciality is " + speciality + ".");
        Thrower thrower = new Thrower(firstName, lastName, speciality);
        thrower.setAthlete(mockAthlete);

        // When
        thrower.display();

        // Then
        assertThat(thrower.getFirstName()).isEqualTo(firstName);
        assertThat(thrower.getLastName()).isEqualTo(lastName);
        verify(mockAthlete).mySpeciality();
    }
}

File: reuse_kotlin/src/test/kotlin/ThrowerTest.kt
import com.kmmania.Athlete
import com.kmmania.Thrower
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.mockito.Mockito.*

class ThrowerTest {
    @Test
    fun `display() method should display correct information`() {
        // Given
        val firstName = "Ryan"
        val lastName = "Crouser"
        val speciality = "shot put"
        val athleteMock = mock(Athlete::class.java)
        `when`(athleteMock.mySpeciality()).thenReturn("My speciality is $speciality.")
        val thrower = Thrower(firstName, lastName, athleteMock)

        // When
        thrower.display()

        // Then
        assertThat(thrower.firstName).isEqualTo(firstName)
        assertThat(thrower.lastName).isEqualTo(lastName)
        verify(athleteMock).mySpeciality()
    }
}